Wikipedia/View Controllers/WebView/NativeSectionTitleOverlays/WMFSectionTitlesViewController.m

```diff
@@ -0,0 +1,231 @@
+                NSLayoutConstraint* c =
+                    [NSLayoutConstraint constraintWithItem:label
+                                                 attribute:attr
+                                                 relatedBy:NSLayoutRelationEqual
+                                                    toItem:browserView
+                                                 attribute:attr
+                                                multiplier:1.0
+                                                  constant:0];
+
+                [self.webView.scrollView addConstraint:c];
+
+                return c;
+            };
+
+            constrainEqually(NSLayoutAttributeTrailing);
+            constrainEqually(NSLayoutAttributeLeading);
+
+            NSLayoutConstraint* topConstraint = constrainEqually(NSLayoutAttributeTop);
+
+            WMFTitleOverlayModel* m = [[WMFTitleOverlayModel alloc] init];
+            m.anchor        = section[@"anchor"];
+            m.title         = title;
+            m.topConstraint = topConstraint;
+            m.yOffset       = topConstraint.constant;
+            m.label         = label;
+            m.sectionId     = sectionId;
+            [self.models addObject:m];
+        }
+    }
+
+    [self updateOverlaysPositions];
+}
+
+- (void)setupTopStaticNativeTitleOverlayLabel {
+    if (self.topStaticNativeTitleLabel) {
+        return;
+    }
+    self.topStaticNativeTitleLabel       = [[WMFTitleOverlayLabel alloc] init];
+    self.topStaticNativeTitleLabel.alpha = 0;
+    [self.webViewController.view addSubview:self.topStaticNativeTitleLabel];
+    [self.topStaticNativeTitleLabel mas_makeConstraints:^(MASConstraintMaker* make) {
+        make.left.equalTo(self.webViewController.view.mas_left);
+        make.right.equalTo(self.webViewController.view.mas_right);
+    }];
+
+    self.topStaticNativeTitleLabelTopConstraint =
+        [NSLayoutConstraint constraintWithItem:self.topStaticNativeTitleLabel
+                                     attribute:NSLayoutAttributeTop
+                                     relatedBy:NSLayoutRelationEqual
+                                        toItem:self.webViewController.topLayoutGuide
+                                     attribute:NSLayoutAttributeBottom
+                                    multiplier:1.0
+                                      constant:0];
+    [self.webViewController.view addConstraint:self.topStaticNativeTitleLabelTopConstraint];
+}
+
+- (void)updateOverlaysPositions {
+    NSArray* headingsTopOffsets = [self getSectionTitlesLocationsJSON];
+    if (headingsTopOffsets.count == self.models.count) {
+        for (NSUInteger i = 0; i < self.models.count; i++) {
+            WMFTitleOverlayModel* m = self.models[i];
+            if (m.anchor && m.topConstraint) {
+                NSNumber* topOffset = headingsTopOffsets[i];
+                CGFloat topFloat    = topOffset.floatValue + self.webView.scrollView.contentOffset.y;
```

Comment:
"+ self.webView.scrollView.contentOffset.y" should be translated to domain knowledge so we understand why this is done.

To increase readability this line could be replaced with: 
```
topOffset.floatValue + [self headerOffSet];

- (CGFloat)headerOffset{
    return self.webView.scrollView.contentOffset.y;
}
```